apps/grid/grid.py

```diff
@@ -71,8 +71,10 @@ class GridBenchmark_ITT(GridBenchmark):
     @run_after('setup')
     def setup_num_tasks(self):
         self.num_tasks = reduce(mul, list(map(int, self.mpi.split("."))), 1)
-        self.set_var_default('num_cpus_per_task',
-                             self.current_partition.processor.num_cpus)
+        self.set_var_default(
+            'num_cpus_per_task',
+            self.current_partition.processor.num_cpus //
+            min(1, self.current_partition.processor.num_cpus_per_core))
```

Comment:
I also did this change suggested by Lokesh the other day during the meeting we had, related to the fact some systems have `current_partition.processor.num_cpus_per_core > 1` when (presumably) hyperthreading is enabled, and related to the fact the new setting of cosma8 correctly sets `num_cpus` _and_ `num_cpus_per_core`.